### PR TITLE
Generate test certificates on the fly

### PR DESCRIPTION
**Pull request by zuiderkwast**
TLS certificates are generated when the TLS tests are started, rather
than having certificates in the repo which can expire.

### CODE REVIEW

**Review by bjosv — APPROVED** [review]
We can remove `priv/update-client-cert.sh` as well now

**Comment by zuiderkwast** [comment]
> We can remove `priv/update-client-cert.sh` as well now

Yes. There seem to be some other problems though. E.g.

```
=ERROR REPORT==== 4-Jun-2021::11:00:47.820150 ===
beam/beam_load.c(1871): Error loading module rebar3:
  This BEAM file was compiled for a later version of the run-time system than 21.
  To fix this, please recompile this module with an 21 compiler.
  (Use of opcode 164; this emulator supports only up to 163.)
```

**Comment by zuiderkwast** [comment]
Rebar3 is compiled for a later OTP version. Do we need to compile rebar ourselves? From the rebar3 README:

    $ git clone https://github.com/erlang/rebar3.git
    $ cd rebar3
    $ ./bootstrap

(This is one of the reasons I don't like rebar... A build tool should either be contained within OTP itself or be completely standalone like Make.)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,13 +27,6 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
```
Comment (bjosv) [inline .github/workflows/ci.yml:29]: If this works we can keep the cache for the hex packages, adding the otp-version to the cache key, and skip using an optional restore-key when the key was not found
```suggestion

      - name: Cache Hex packages
        uses: actions/cache@v2
        with:
          path: ~/.cache/rebar3/hex/hexpm/packages
          key: ${{ runner.os }}-${{ matrix.otp-version }}-hex-${{ hashFiles('**/rebar.lock') }}
```

Comment (bjosv) [reply .github/workflows/ci.yml:29]: This should avoid using the the same built packages for all otp-versions

**Pull request merged**